### PR TITLE
getting library list from source

### PR DESCRIPTION
**Pull request by zhumin8**
Use `google-cloud-java` as starting point to process list of client libraries and related info needed for generation.
Start point:  a `google-cloud-java` release tag.
Parsed info to use for generation:
- client library maven group id and artifact id
- googleapis folder to use as source of truth
- googleapis latest committish before tagged release with changes to this client library.

Per conversation with @suztomo , use `.OwlBot.yaml` and git log as source of truth for folder location and committish.
Major steps:
Loops through `java-*` folders in `google-cloud-java`, for each client library:
 - parse variables from `.repo-metadata.json`, use as criteria to filter in-scope library list: #1322 
 - (updated) remove ones that manual module exists (vision, kms)
 - get folder location "source of truth" from `.OwlBot.yaml`
     - get folder location where `/.*-java/gapic-google-.*/` is copied over. Note: this is technically `googleapis-gen` folder location, but copied from same folder in `googleapis`
     - also get dest folder name in `google-cloud-java`, this will be used to build path to look for git log change.
     - find the version to generate. (we only generate one module per client library for stable version, skipping beta versions. (updated)**Taking the latest stable version if there are multiple**.)
 - get commitish from git log
     - get the last git change for path concerning this client library

This method is subject to any structural/process change in `google-cloud-java`. For longer term, consider maven artifact as the only source of truth. But for the current phase, this heuristic method is needed to grab all information needed.

Todo: 
~~- add v1/v2 to folder info.~~
~~- remove getting client library versions. (Not needed anymore. Generated code will get this version from Library BOM)~~

### CODE REVIEW

**Review by meltsufin — COMMENTED** [review]
LGTM

**Comment by zhumin8** [comment]
Update: 
- Remove ones that manual module exists (vision, kms)
- Add version info to folder location string in library_list.txt
- Taking the latest stable version if there are multiple

File: generator/generate-library-list.sh [hunk 1]
```diff
@@ -60,36 +61,40 @@ for d in ./google-cloud-java/*java-*/; do
...
+  # taking the latest version that's not alpha/beta
+  version_folder=$(find "$d$repo_folder/main/java/com/google/" -type d -name 'v[0-9]' |sort -d -r | head -n 1 | sed "s#^$d##")
+  version_number=$(echo $version_folder | sed 's#.*/##')
+  googleapis_folder="$googleapis_path/$version_number"
 
-  # get version from versions.txt
+  # get commitish from git log
+  # criteria: changes happen before tag,  touches path, and with changes in googleapis/googleapis
+  cd $d || { echo "Failed to get into directory $d"; exit 1; }
+  googleapis_committish=$(git log v1.1.0 -- "$version_folder" | grep -m 1 'Source-Link:.*googleapis/googleapis.*' | sed 's#^.*/commit/##')
```
Comment (emmileaf) [inline generator/generate-library-list.sh:85]: Here I assume `v.1.1.0` is meant to be the flag argument instead?

Comment (zhumin8) [reply generator/generate-library-list.sh:85]: yes! good catch.

**Review by emmileaf — APPROVED** [review]

**Pull request merged**